pkg/infrastructure/gcp/clusterapi/clusterapi.go

```diff
@@ -170,6 +176,20 @@ func (p Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput)
 		return fmt.Errorf("failed to get GCP cluster: %w", err)
 	}
 
+	if in.InstallConfig.Config.GCP.UserProvisionedDNS != dns.UserProvisionedDNSEnabled {
+		if in.InstallConfig.Config.GCP.Endpoint != nil {
```

Comment:
Should we have a static to make sure that the 2 features (custom dns vs custom endpoint) are mutually exclusive like Patrick said: https://github.com/openshift/installer/pull/9992#pullrequestreview-3298959304